src/main/java/seedu/address/logic/parser/gradetrackerparsers/DeleteAssignmentParser.java

```diff
@@ -1,12 +1,34 @@
 package seedu.address.logic.parser.gradetrackerparsers;
 
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+
+import seedu.address.commons.core.index.Index;
 import seedu.address.logic.commands.gradetrackercommands.DeleteAssignmentCommand;
+import seedu.address.logic.parser.ArgumentMultimap;
+import seedu.address.logic.parser.ArgumentTokenizer;
 import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.ParserUtil;
 import seedu.address.logic.parser.exceptions.ParseException;
 
 public class DeleteAssignmentParser implements Parser<DeleteAssignmentCommand> {
+
     @Override
-    public DeleteAssignmentCommand parse(String userInput) throws ParseException {
-        return null;
+    public DeleteAssignmentCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        String moduleName;
+        Index index;
+
+        ArgumentTokenizer tokenizer = new ArgumentTokenizer(args, PREFIX_NAME);
+        ArgumentMultimap argMultimap = tokenizer.tokenize();
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    DeleteAssignmentCommand.MESSAGE_USAGE), pe);
+        }
+        moduleName = ParserUtil.parseModuleName(argMultimap.getValue(PREFIX_NAME).get()).fullName;
```

Comment:
Since parseModuleName returns a ModuleName object, you could consider using a ModuleName object rather than a String for the argument of DeleteAssignmentCommand. This is a continuation from the comment on the DeleteAssignmentCommand class. Saving the moduleName variable as a ModuleName object makes it easier to compare module names.